gazebo/create_dockerfiles.py

```diff
@@ -16,6 +19,35 @@
 from ros_buildfarm.docker_common import OrderedLoad
 
 
+def checkVersion(data):
+    """Checks current package_version from online package index"""
+
+    # Determine URL
+    urlTemplate = string.Template(
+        """
+        http://packages.osrfoundation.org/gazebo/$os_name/dists/$os_code_name/main/binary-$arch/Packages
+        """)
+    url = urlTemplate.substitute(data)
+
+    # Download package index
+    req = urllib.request.Request(url)
+    with urllib.request.urlopen(req) as response:
+        package_index = response.read().decode('utf-8')
+
+    # Determine seach pattern
+    patternTemplate = string.Template(r'(\bPackage: gazebo$version\n)(.*\n)')
+    pattern_raw = patternTemplate.substitute(data)
+    pattern = re.compile(pattern_raw)
+
+    # Parse for version_number
+    matchs = re.search(pattern, package_index)
+    version_line = matchs.groups(0)[1] # Grab the second line of the first match
+    version_number = re.search(r'\d{1}[.]\d{1}[.]\d{1}', version_line).group(0) # extract version_number
```

Comment:
I'm a little rusty on my regex, whats a good way to grab that, up to the second tilda?
Would this cover all your cases?
`\d{1}[.]\d{1}[.]\d{1}~\w*~`
or
`((\d)+[.])+\d+((~|\+)[\w,\-]*(?=~|\n)|\n)`

The last one should match with stuff like:
`127.0.0.1~beta9000~spam\n` -> `127.0.0.1~beta9000`
`127.0.0.1~beta9000\n` -> `127.0.0.1~beta9000`
`127.0.0\n` -> `127.0.0`
I'm not sure of all numeric conventions there are in package versioning format constraints?
Theres a wide variaty of stuff like stuff like [this](http://packages.osrfoundation.org/gazebo/ubuntu-stable/dists/trusty/main/binary-amd64/Packages):
`Version: 0.0.0+git20170131r43c21ae5115b693ec0edc551abbc189a54e3c2ab-4~trusty` -> `0.0.0+git20170131r43c21ae5115b693ec0edc551abbc189a54e3c2ab-4`

FYI: you can debug them with [regexr.com](http://regexr.com/) or [debuggex.com](https://www.debuggex.com/)